packages/react-scheduler/src/ReactScheduler.js

```diff
@@ -190,11 +201,19 @@ if (!ExecutionEnvironment.canUseDOM) {
     callback: (deadline: Deadline) => void,
     options?: {timeout: number},
   ): number {
-    // This assumes that we only schedule one callback at a time because that's
-    // how Fiber uses it.
+    if (scheduledRICCallback !== null) {
+      // Handling multiple callbacks:
+      // For now we implement the behavior expected when the callbacks are
+      // serial updates, such that each update relies on the previous ones
+      // having been called before it runs.
+      const didTimeout = (timeoutTime !== -1) && (timeoutTime <= now());
+      scheduledRICCallback(buildFrameDeadlineObject(didTimeout));
```

Comment:
It's common for callbacks to reschedule themselves if time runs out in the frame and they have remaining work. (The exception is callbacks that ignore the deadline object entirely.) I don't think this implementation accounts for that case.

Example:

```js
function callbackB() {}

function callbackA(deadline) {
  while (hasWork()) {
    if (deadline.didTimeout || deadline.timeRemaining() < 1) {
      // Ran out of time. Reschedule self so we can continue later.
      rIC(callbackA);
    }
    doUnitOfWork();
  }
}

// Schedule a callback
rIC(callbackA);
// With your PR, this will early flush callback A...
// If callback A runs out of time, it will reschedule itself, all before we get a chance to schedule callback B.
rIC(callbackB);
```

Reply:
Good point!

I will write some tests around this and take it into account.